bin/github-changelog.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+require __DIR__ . '/../vendor/erusev/parsedown/Parsedown.php';
+
+define( 'GITHUB_ENDPOINT', 'https://api.github.com/repos/Automattic/vip-go-mu-plugins/pulls?per_page=10&sort=updated&direction=desc&state=closed');
+define( 'WP_CHANGELOG_ENDPOINT', 'https://public-api.wordpress.com/wp/v2/sites/wpvipchangelog.wordpress.com/posts');
+define( 'WP_CHANGELOG_AUTH_TOKEN', $_SERVER['CHANGELOG_POST_TOKEN'] );
+define( 'PR_CHANGELOG_HEADER', '<h2>Changelog Description' );
+define( 'PR_DESCRIPTION_HEADER_PREFIX', '<h2>' );
+
+function fetch_last_PR() {
+    $ch = curl_init( GITHUB_ENDPOINT );
+    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+    curl_setopt($ch, CURLOPT_HEADER, 0);
+    curl_setopt($ch, CURLOPT_HTTPHEADER, [ 'User-Agent: script' ] );
+    $data = curl_exec($ch);
+    curl_close($ch);
+
+    return json_decode($data, true)[0];
+}
+
+function get_changelog_section_in_description_html( $description ) {
+    $found_changelog_header = false;
+    $result = '';
+    foreach(preg_split("/\n/", $description) as $line){
+        if ( strpos($line, PR_CHANGELOG_HEADER) === 0 ) {
+            $found_changelog_header = true;
+        } else if ( $found_changelog_header ) {
+
+            if ( strpos($line, PR_DESCRIPTION_HEADER_PREFIX) === 0 ) {
+                // We have hit next section
+                break;
+            }
+            $result = $result . "\n" . $line;
+        }
+    }
+    return $result;
+}
+
+function get_changelog_html( $pr ) {
+    $Parsedown = new Parsedown();
+    $description_html =  $Parsedown->text( $pr['body'] );
```

Comment:
```suggestion
    $body = preg_replace( '/<!--(.|\s)*?-->/', '', $pr['body'] );
    $description_html =  $Parsedown->text( $body );
```
I suggest removing HTML comments from the PR description (some PRs are keeping the original comments in the default description), to avoid them reaching the changelog post.

Reply:
Cool yeah, I noticed though... but didn't trust my regexp skills enough:)